internal/appsec/dyngo/instrumentation/httpsec/http.go

```diff
@@ -128,37 +122,46 @@ func WrapHandler(handler http.Handler, span ddtrace.Span, pathParams map[string]
 		ipTags, clientIP := ClientIPTags(r.Header, true, r.RemoteAddr)
 		instrumentation.SetStringTags(span, ipTags)
 
+		var bypassHandler http.Handler
 		args := MakeHandlerOperationArgs(r, clientIP, pathParams)
-		ctx, op := StartOperation(r.Context(), args)
+		op := NewOperation(nil)
+		sharedsec.OnData(op, func(a *sharedsec.Action) {
+			bypassHandler = a.HTTP()
+			if a.Blocking() {
+				op.AddTag(instrumentation.BlockedRequestTag, true)
+			}
+		})
+		ctx := StartOperation(r.Context(), op, args)
 		r = r.WithContext(ctx)
 
-		if h := applyActions(op); h != nil {
-			handler = h
-		}
 		defer func() {
 			var status int
 			if mw, ok := w.(interface{ Status() int }); ok {
 				status = mw.Status()
 			}
 
 			events := op.Finish(HandlerOperationRes{Status: status})
-			if h := applyActions(op); h != nil {
-				h.ServeHTTP(w, r)
-			}
 			// Execute the onBlock functions to make sure blocking works properly
 			// in case we are instrumenting the Gin framework
 			if _, ok := op.Tags()[instrumentation.BlockedRequestTag]; ok {
 				for _, f := range onBlock {
 					f()
 				}
 			}
+			if bypassHandler != nil {
+				bypassHandler.ServeHTTP(w, r)
+			}
```

Comment:
Because of that block, where the handler wasn't bypassed, I suggest you rename `bypassHandler` into `blockingHandler`.

Reply:
It could be a redirecting handler too, which is why bypass seemed right since the end result in both cases is the user handler being bypassed.